transaction/core/src/ring_signature/rct_bulletproofs.rs

```diff
@@ -394,7 +394,7 @@ fn compute_extended_message_either_version(
     pseudo_output_commitments: &[CompressedCommitment],
     range_proof_bytes: &[u8],
 ) -> Vec<u8> {
-    if block_version >= BlockVersion::THREE {
+    if block_version >= BlockVersion::TWO {
```

Comment:
Shouldn't this use the `BlockVersion::has_feature()` API?

Reply:
it's not exactly the same as the token id feature

this is basically, there was a lack of proper framing and domain separation when creating the "message" that is signed by the MLSAG, and I wanted to fix it.

we could add a new feature like "fix extended message framing" and make it contingent on that I guess. that seems fine to me. lmk what you want

Reply:
I think any fix should be in a follow up PR, this is already too big